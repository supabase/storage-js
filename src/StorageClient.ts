import { StorageBucketApi, StorageFileApi } from './lib'
import { Fetch } from './lib/fetch'

export class StorageClient extends StorageBucketApi {
  constructor(url: string, headers: { [key: string]: string } = {}, fetch?: Fetch) {
    super(url, headers, fetch)
  }

  /**
   * Perform file operation in a bucket.
   *
   * @param id The bucket id to operate on.
   */
  from(id: string): StorageFileApi {
    return new StorageFileApi(this.url, this.headers, id, this.fetch)
  }
}
