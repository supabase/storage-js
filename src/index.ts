export {
  StorageClient as StorageClient,
  StorageClient as SupabaseStorageClient,
} from './StorageClient'
export * from './lib/types'
