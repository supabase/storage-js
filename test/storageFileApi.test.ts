import { StorageClient } from '../src/index'

// TODO: need to setup storage-api server for this test
const URL = 'http://localhost:8000/storage/v1'
const KEY =
  'eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJzdXBhYmFzZSIsImlhdCI6MTYwMzk2ODgzNCwiZXhwIjoyNTUwNjUzNjM0LCJhdWQiOiIiLCJzdWIiOiIzMTdlYWRjZS02MzFhLTQ0MjktYTBiYi1mMTlhN2E1MTdiNGEiLCJSb2xlIjoicG9zdGdyZXMifQ.pZobPtp6gDcX0UbzMmG3FHSlg4m4Q-22tKtGWalOrNo'

const storage = new StorageClient(URL, { Authorization: `Bearer ${KEY}` })
const newBucketName = 'my-new-public-bucket'

test('get public URL', async () => {
  const res = storage.from(newBucketName).getPublicUrl('profiles/myUniqueUserId/profile.png')
  expect(res.error).toBeNull()
  expect(res.data).toMatchSnapshot()
  expect(res.publicURL).toMatchSnapshot()
})
