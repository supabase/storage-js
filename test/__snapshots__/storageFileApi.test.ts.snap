// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`get public URL 1`] = `"http://localhost:8000/storage/v1/object/public/my-new-public-bucket/profiles/myUniqueUserId/profile.png"`;
